### PR TITLE
Fixes and more features

### PR DESCRIPTION
**Pull request by ThisTestUser**
Full list of features:
- Added customlist targeting mode
- Added /botenv for creating custom list of targets
- Added support in /botenv for modded Magma servers
- Fixed target option PLAYER
- Fixed autofill bugs and add aliases in autofill
- Fixed bot not appearing in tablist when addplayerlist enabled
- Fixed held items sometimes not updating (for example totems)
- Sound retrieval fixes
- Better tool switching (cosmetic)
- Extra /bot debug commands and autofill

/botenv contains some code that only works for Magma (https://github.com/magmamaintained/Magma-1.20.1) servers. Since that code is accessed using reflection, no issues will occur outside of Magma.

The magma branch referenced in the second commit is at https://github.com/ThisTestUser/TerminatorPlus/tree/1.18.2-magma.

### CODE REVIEW

**Comment by Badbird5907** [comment]
:eyes: I'll have a look at this PR in a bit (a couple of hours), please ping me on discord tonight if I forget.

File: TerminatorPlus-Plugin/src/main/java/net/nuggetmc/tplus/bot/Bot.java [hunk 1]
```diff
@@ -93,6 +93,11 @@ private Bot(MinecraftServer minecraftServer, ServerLevel worldServer, GameProfil
             inPlayerList = true;
         }
 
+        // Magma fix - In case a mod causes the Bukkit entity to be created too early
```
Comment (Badbird5907) [inline TerminatorPlus-Plugin/src/main/java/net/nuggetmc/tplus/bot/Bot.java:96]: I personally don't like supporting hybrid servers, but sure.

File: TerminatorPlus-Plugin/src/main/java/net/nuggetmc/tplus/command/commands/BotEnvironmentCommand.java [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+            default:
+                return 0;
+        }
+    }
+
+    private boolean isLocationLoaded(Location loc) {
+        return loc.getWorld().isChunkLoaded(Location.locToBlock(loc.getX()) >> 4, Location.locToBlock(loc.getZ()) >> 4);
+    }
+
+    private static Set<String> getLoadedMods() {
```
Comment (Badbird5907) [inline TerminatorPlus-Plugin/src/main/java/net/nuggetmc/tplus/command/commands/BotEnvironmentCommand.java:407]: Hybrid support

**Review by Badbird5907 — CHANGES_REQUESTED** [review]
LGTM

**Comment by Badbird5907** [comment]
On second thought, could you please move the hybrid server support to a fork of this repo?
I won't be maintaining nor testing the support for hybrid servers, and I think it would be easier to maintain your own fork with those features and pull in patches from upstream.

**Comment by ThisTestUser** [comment]
All code referencing hybrid server classes have been removed. I've kept /botenv addSolid as that only uses Bukkit/Vanilla methods.

**Review by Badbird5907 — APPROVED** [review]
LGTM!

Thank You!

**Pull request merged**